lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1255,9 +1252,16 @@ matchWitness(TypeChecker &tc,
   };
 
   // Match a type in the requirement to a type in the witness.
-  auto matchTypes = [&](Type reqType, Type witnessType) 
+  auto matchTypes = [&](Type reqType, Type witnessType)
                       -> Optional<RequirementMatch> {
-    cs->addConstraint(ConstraintKind::Equal, reqType, witnessType, locator);
+    ConstraintKind kind = ConstraintKind::Conversion;
+    if (auto func = dyn_cast<FuncDecl>(req)) {
+      auto name = req->getFullName().getBaseName();
+      if (func->isOperator() && name.str() == "==") {
```

Comment:
Why is this restriction specifically in place?  Going by name alone isn't going to only pick out Equatable requirements.